includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -250,9 +259,9 @@ public static function is_wcpay_subscription( WC_Subscription $subscription ) :
 	 */
 	public static function format_item_price_data( string $currency, string $wcpay_product_id, float $unit_amount, string $interval = '', int $interval_count = 0 ) : array {
 		$data = [
-			'currency'    => $currency,
-			'product'     => $wcpay_product_id,
-			'unit_amount' => (int) $unit_amount * 100,
+			'currency'            => $currency,
+			'product'             => $wcpay_product_id,
+			'unit_amount_decimal' => $unit_amount * 100,
```

Comment:
The prior code was causing a rounding error.

e.g. I have a subscription that includes a line item of 3 x subscription products for a total of $100.

Each individual line item would equal $33.333333333333.

The prior code would do this:

```php
(int) 33.333333333333 * 100
// int(3300)
```

Which would result in an invoice for $99 (a $1 discount for the expected invoice total of $100)

The new code does this:

```php
33.333333333333 * 100
// float(3333.3333333333)
```

The line items in Stripe equal $33.333333333333 but the final invoice comes out to a total of $100 as expected.

Reply:
Oh, I see now. You also switched to `unit_amount_decimal` which explains why this works. 😆

Reply:
Yeah, sorry, I should've mentioned that! 😅